hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/TokenContextImpl.java

```diff
@@ -148,8 +143,6 @@ public void markMigrationRecordsStreamed() {
 
     @Override
     public Set<Long> knownNodeIds() {
-        return new ReadableStoreFactory(state)
-                .getStore(ReadableStakingInfoStore.class)
-                .getAll();
+        return storeFactory.readableStore(ReadableStakingInfoStore.class).getAll();
```

Comment:
This looks like we just missed passing the stack into `ReadableStoreFactory`'s constructor here, but it may still be worth asking: are there ever any differences between the state object and the stack passed in to `storeFactory`? For example, in the case of a dispatched child transaction

Reply:
Yes. If a new savepoint is created and subsequently modified, the changes are visible to the stack but not to the root state